.github/workflows/ci.yml

```diff
@@ -0,0 +1,122 @@
+name: Build and push
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  base-images:
+    # we cannot offer this caching optimizations to forks as it would require access to our Dockerhub
+    if: github.repository == 'pgautoupgrade/docker-pgautoupgrade'
```

Comment:
I'm note sure if we have an attack possibility here:

1. Somebody creates a PR. Since they are a first time contributor, we need to approve that the action runs.
2. We review the code, all fine, change gets merged.
3. The contributor creates a second PR with malicious changes injected into these build containers. Since they were approved the first time, the workflow now runs automatically, which pushes new build containers.
4. We rebuild the main images on the main branch, which pull the updated build containers with potential harmful changes.

I'm not sure, given a PR from a fork, what `github.repository` will be.

we could prevent this in multiple ways:

- Build specific build containers per branch.
- Always require an approval to run workflows when committing from a fork.
- Only build these build containers on main (and have longer build times on feature branches).

Reply:
That particular worry is an open question for OSS in general.

That being said, you already have what seems like a reasonable + secure possibility in the list:

> Only build these build containers on main (and have longer build times on feature branches).

As long as the builds complete within the 6 hour GitHub Action limit, I personally think that'd be ok.  Hopefully. :smile:

Reply:
addressed with 106b9ac3.

Reply:
Awesome. :smile: